scormxblock/static/js/src/scormxblock.js

```diff
@@ -72,28 +72,46 @@ function ScormXBlock(runtime, element, settings) {
     return response.value
   };
 
+  var handlerUrl = runtime.handlerUrl( element, 'scorm_set_value');
+  var eventsQueue = []
+  var queueRunningState = false
+
   var SetValue = function (cmi_element, value) {
     if (cmi_element === 'cmi.core.exit' || cmi_element === 'cmi.exit') {
       $(".js-scorm-block", element).removeClass('full-screen-scorm');
     }
+    eventsQueue.push({name: cmi_element, value: value})
+    if(!queueRunningState)
+    {
+      processQueueEvents();
+    }
+    return "true";
+  };
 
-    var handlerUrl = runtime.handlerUrl( element, 'scorm_set_value');
-
+  function processQueueEvents ()
+  {
+    if (eventsQueue.length === 0) {
```

Comment:
@tehreem-sadat simply removing async: false makes event API call asynchronous why we need to build a queue?

Reply:
Yeah It was working, but I just added queue to maintain events order.